search/includes/classes/commands/class-versioncommand.php

```diff
@@ -288,67 +355,68 @@ public function delete( $args, $assoc_args ) {
 			return WP_CLI::error( sprintf( 'Indexable %s not found. Is the feature active?', $type ) );
 		}
 	
-		$version_number = $search->versioning->normalize_version_number( $indexable, $args[1] );
+		if ( isset( $assoc_args['network-wide'] ) && is_multisite() ) {
+			if ( ! is_numeric( $assoc_args['network-wide'] ) ) {
+				$assoc_args['network-wide'] = 0;
+			}
 
-		if ( is_wp_error( $version_number ) ) {
-			return WP_CLI::error( sprintf( 'Index version %s is not valid: %s', $args[1], $version_number->get_error_message() ) );
-		}
+			$sites = \ElasticPress\Utils\get_sites( $assoc_args['network-wide'] );
 
-		$active_version_number = $search->versioning->get_active_version_number( $indexable );
+			CoreCommand::confirm_destructive_operation( $assoc_args );
```

Comment:
Minor - to keep things more DRY, can the confirmations go outside the network-wide conditional?

Reply:
Sure! I moved it here because for the single site mode it asks for confirmation after validating the version. But in network mode it has to be done early. So for simplicity it's better to move it outside as you suggest